orderly_web/deploy.py

```diff
@@ -0,0 +1,103 @@
+import docker
+
+from orderly_web.docker_helpers import docker_client, \
+    ensure_network, ensure_volume, \
+    exec_safely, string_into_container
+
+
+def deploy(cfg):
+    with docker_client() as cl:
+        ensure_network(cl, cfg.network)
+        for v in cfg.volumes.values():
+            ensure_volume(cl, v)
+        orderly = orderly_init(cfg, cl)
+        web = web_init(cfg, cl)
+
+
+def orderly_init(cfg, docker_client):
+    container = orderly_container(cfg, docker_client)
+    if not orderly_is_initialised(container):
+        orderly_init_demo(container)
+    orderly_check_schema(container)
+    orderly_start(container)
+    return container
+
+
+def orderly_container(cfg, docker_client):
+    print("Creating orderly container")
+    args = ["--port", "8321", "--go-signal", "/go_signal", "/orderly"]
+    mounts = [docker.types.Mount("/orderly", cfg.volumes["orderly"])]
+    container = docker_client.containers.run(
+        cfg.orderly_image, args, mounts=mounts, network=cfg.network,
+        name=cfg.containers["orderly"], working_dir="/orderly", detach=True)
+    return container
+
+
+def orderly_init_demo(container):
+    print("Initialising orderly with demo data")
+    args = ["Rscript", "-e", "orderly:::create_orderly_demo('/orderly')"]
+    exec_safely(container, args)
+
+
+def orderly_is_initialised(container):
+    res = container.exec_run(["stat", "/orderly/orderly_config.yml"])
+    return res[0] == 0
+
+
+def orderly_check_schema(container):
+    print("Checking orderly schema is current")
+    exec_safely(container, ["orderly", "rebuild", "--if-schema-changed"])
+
+
+def orderly_start(container):
+    print("Starting orderly server")
+    exec_safely(container, ["touch", "/go_signal"])
+
+
+def web_init(cfg, docker_client):
+    container = web_container(cfg, docker_client)
+    web_container_config(cfg, container)
+    web_migrate(cfg, docker_client)
+    web_start(container)
+    return container
+
+
+def web_container(cfg, docker_client):
+    print("Creating web container")
+    image = "vimc/orderly-web:master"
```

Comment:
Should this be configurable, as the orderly.server one is? Similarly the migrations image https://github.com/vimc/orderly-web-deploy/pull/2/files#diff-056cb8e1a08536e35b0f910ea7fba570R96

Reply:
Yes, but this is something I want to tackle outside of this PR, as it breaks things and interacts with VIMC-2803

Reply:
On second thoughts, it makes sense to put the framework for this in (593fef7)